lib/smart_todo/slack_client.rb

```diff
@@ -67,27 +70,19 @@ def post_message(channel, text)
     #
     # @raise [Net::HTTPError] in case the request to Slack failed
     # @raise [SlackClient::Error] in case Slack returns a { ok: false } in the body
-    def request(method, endpoint, data = nil, headers = {})
-      response = case method
-      when :post, :patch
-        @client.public_send(method, endpoint, data, default_headers.merge(headers))
-      else
-        @client.public_send(method, endpoint, default_headers.merge(headers))
-      end
+    def dispatch(request)
+      response = @client.request(request)
+      attempts = 1
 
-      slack_response!(response)
-    end
+      while response.is_a?(Net::HTTPTooManyRequests) && attempts < 5
+        sleep(Integer(response["Retry-After"]))
```

Comment:
To be real safe, and guard against some eccentric response or parsed value, you could do:

```ruby
sleep(min(Integer(response["Retry-After"]), SOME_SAFE_MAXIMUM_SECONDS))
```